src/openrct2-ui/windows/EditorObjectSelection.cpp

```diff
@@ -464,8 +464,8 @@ static void window_editor_object_selection_mouseup(rct_window* w, rct_widgetinde
         case WIDX_CLOSE:
             if (gScreenFlags & SCREEN_FLAGS_EDITOR)
             {
-                auto loadOrQuitAction = LoadOrQuitAction(LoadOrQuitModes::OpenSavePrompt, PM_SAVE_BEFORE_QUIT);
-                GameActions::Execute(&loadOrQuitAction);
+                window_close(w);
```

Comment:
`window_close(w)` is called in both the if and the else clause. Move it before the if and remove the else clause.